crawlera_fetch/middleware.py

```diff
@@ -193,11 +262,17 @@ def process_response(self, request: Request, response: Response, spider: Spider)
                 response.status,
                 message,
             )
-            if self.raise_on_error:
+            if self.on_error_action == OnError.Raise:
                 raise CrawleraFetchException(log_msg)
-            else:
+            elif self.on_error_action == OnError.Warn:
                 logger.warning(log_msg)
                 return response
+            elif self.on_error_action == OnError.Retry:
+                return self._get_retry_request(
+                    request=request,
+                    reason=message,
+                    stats_base_key="crawlera_fetch/retry/error",
+                )
```

Comment:
Maybe we should use `else` + `assert` here, or have an `else` that raises `NotImplementedError`, just to avoid an programming error in the future causing an unexpected, possibly hard to catch behavior here. Similar below.

Reply:
What about c77983ee4a5716548eddc03ae70c3718226d6e02?